src/Evolution/Systems/Cce/AnalyticSolutions/RobinsonTrautman.cpp

```diff
@@ -0,0 +1,404 @@
+    const double extraction_radius, const size_t l_max, const double tolerance,
+    const double start_time) noexcept
+    : SphericalMetricData{extraction_radius},
+      dense_output_rt_scalar_{
+          Spectral::Swsh::number_of_swsh_collocation_points(l_max)},
+      dense_output_du_rt_scalar_{
+          Spectral::Swsh::number_of_swsh_collocation_points(l_max)},
+      l_max_{l_max},
+      tolerance_{tolerance},
+      start_time_{start_time},
+      initial_modes_{std::move(initial_modes)} {
+  initialize_stepper_from_start();
+}
+
+void RobinsonTrautman::initialize_stepper_from_start() noexcept {
+  // create the initial data
+  SpinWeighted<ComplexModalVector, 0> goldberg_modes{square(l_max_ + 1), 0.0};
+  for (size_t i = 0; i < std::min(initial_modes_.size(), goldberg_modes.size());
+       ++i) {
+    goldberg_modes.data()[i] = gsl::at(initial_modes_, i);
+  }
+
+  SpinWeighted<ComplexModalVector, 0> libsharp_modes{
+      Spectral::Swsh::size_of_libsharp_coefficient_vector(l_max_)};
+  const auto& coefficients_metadata =
+      Spectral::Swsh::cached_coefficients_metadata(l_max_);
+  for (const auto& coefficient : coefficients_metadata) {
+    Spectral::Swsh::goldberg_modes_to_libsharp_modes_single_pair(
+        coefficient, make_not_null(&libsharp_modes), 0,
+        goldberg_modes.data()[Spectral::Swsh::goldberg_mode_index(
+            l_max_, coefficient.l, static_cast<int>(coefficient.m), 0)],
+        goldberg_modes.data()[Spectral::Swsh::goldberg_mode_index(
+            l_max_, coefficient.l, -static_cast<int>(coefficient.m), 0)]);
+  }
+
+  // re-use the `dense_output_rt_scalar_` buffers to reduce allocations
+  Spectral::Swsh::inverse_swsh_transform(
+      l_max_, 1, make_not_null(&get(dense_output_rt_scalar_)), libsharp_modes);
+  // offset the scalar to 'reasonable' value
+  get(dense_output_rt_scalar_).data() =
+      1.0 + std::complex<double>(1.0, 0.0) *
+                real(get(dense_output_rt_scalar_).data());
+  du_rt_scalar(make_not_null(&get(dense_output_du_rt_scalar_)),
+               get(dense_output_rt_scalar_));
+  stepper_ = boost::numeric::odeint::make_dense_output(
+      tolerance_ * 0.01, tolerance_,
+      boost::numeric::odeint::runge_kutta_dopri5<ComplexDataVector>{});
+  const double rt_scalar_scale = max(abs(get(dense_output_rt_scalar_).data()));
+  const double du_rt_scalar_scale =
+      max(abs(get(dense_output_du_rt_scalar_).data()));
+
+  double initial_step = extraction_radius_ * 1.0e-4 / square(l_max_);
+  if (rt_scalar_scale > 1.0e-5 and du_rt_scalar_scale > 1.0e-5) {
+    initial_step =
+        0.1 * rt_scalar_scale / (du_rt_scalar_scale * square(l_max_));
+  }
+  stepper_.initialize(get(dense_output_rt_scalar_).data(), start_time_,
+                      initial_step);
+
+  const auto rt_system = [this](const ComplexDataVector& local_rt_scalar,
+                                ComplexDataVector& local_du_rt_scalar,
+                                const double /*t*/) noexcept {
```

Comment:
I think this could be a private member function so you can re-use it in the time-stepping below.

Reply:
Unfortunately, this doesn't work, because boost needs a callable passed to its system argument and a non-static member function isn't directly callable -- it would need to use the object at the call site, I think.